sklearn/ensemble/gradient_boosting.py

```diff
@@ -1836,3 +1843,353 @@ def apply(self, X):
+              spawning of the jobs
+
+            - An int, giving the exact number of total jobs that are
+              spawned
+
+            - A string, giving an expression as a function of n_jobs,
+              as in '2*n_jobs'
+
+    verbose : integer
+        Controls the verbosity: the higher, the more messages.
+
+    Attributes
+    ----------
+    grid_scores_ : list of named tuples
+        Contains scores for all parameter combinations in param_grid.
+        Each entry corresponds to one parameter setting.
+        Each named tuple has the attributes:
+
+            * ``parameters``, a dict of parameter settings
+            * ``mean_validation_score``, the mean score over the
+              cross-validation folds
+            * ``cv_validation_scores``, the list of scores for each fold
+
+    best_score_tuple_ : named tuple
+        The attributes of the model which gave the best score.
+
+        * ``parameters``, a dict of parameter settings, which includes the
+          `n_estimators` parameter
+        * ``mean_validation_score``, the mean score over the
+          cross-validation folds
+        * ``cv_validation_scores``, the list of scores for each fold
+
+    best_params_ : dict
+        The parameters which gave the best score, along with the `n_estimators`
+        parameter.
+
+    best_estimator_ : GradientBoostingClassifier
+        If `refit` was `True`, the model with the best score fit to the entire
+        dataset.
+
+    """
+
+    estimator_class = GradientBoostingClassifier
+
+    def __init__(self, **kwargs):
+
+        self.n_stop_rounds = kwargs.pop('n_stop_rounds', 10)
+        self.score_precision = kwargs.pop('score_precision', 2)
+        self.max_iterations = kwargs.pop('max_iterations', 10000)
+        self.cv = kwargs.pop('cv', 3)
+        self.scoring = kwargs.pop('scoring', None)
+        self.refit = kwargs.pop('refit', True)
+        self.n_jobs = kwargs.pop('n_jobs', 1)
+        self.pre_dispatch = kwargs.pop('pre_dispatch', '2*n_jobs')
+        self.verbose = kwargs.pop('verbose', False)
+        self._params = kwargs
+
+        # Pre processing to ensure every parameter is a list
+        # which `ParameterGrid` can iterate over
+        for key, value in self._params.items():
+            if not type(value) is list:
+                self._params[key] = [value]
+
+    def fit(self, X, y):
+        """Run fit with all sets of parameters till convergence.
+
+        X : array-like, shape = [n_samples, n_features]
+            Training vectors, where n_samples is the number of samples
+            and n_features is the number of features.
+        y : array-like, shape = [n_samples]
+            Target values (integers in classification, real numbers in
+            regression)
+            For classification, labels must correspond to classes.
+        """
+
+        X, y = check_X_y(X, y, dtype=DTYPE)
+
+        cv = check_cv(self.cv, y)
+        param_iter = ParameterGrid(self._params)
```

Comment:
this isn't needed. See the base module of ensemble if you want to do something more sophisticated.